### ISSUE TITLE
help & [variation]s: effect of the hide_help key (GNA #25624)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1647]
**[Original submission](https://gna.org/bugs/index.php?25624)** by shiki on 2017-03-23

(Reproduced on Archlinux)
Release: 1.13.7+dev
Priority: 5 - Normal
Severity: 3 - Normal

**Comment by wesnoth-bugs** [comment]
**Modified on 2017-03-23**

**shiki wrote:**

> help & [variation]s: effect of the hide_help key
> 
> 
> 
> Imagine you have one unit with one variation. Like Konrad in HttT.
> 
> And imagine you don't use inherit=yes (unlike the zombies), not sure if that would make a difference.
> 
> 
> 
> Currently Konrad and Li'sar use hide_help=yes for the variation, and not for the main unit.
> 
> This has three effects
> 
>   -> 1) the variation is not shown in the help browser
> 
>   -> 2) the variation is not shown in the main units wiki page. There would usually be a line "variations:"
> 
>   -> 3) when the variation is in effect, e.g. Konrad has the sceptre, it shows the page of the main unit instead of the variation.
> 
> This is different than in the usual cases where hide_help#yes it used. As a rule of thumb, hide_helpyes does hide the unit in the help system, but there exists still a help page which is only accessible if the unit is on the field.
> 
> tl;dr, Konrads Sceptre attack isn't displayed in the help when he has it.
> 
> 
> 
> Why did the designer decide for that? We'll, let's have a look at the other options:
> 
> 
> 
> Using hide_help=yes for both:
> 
>   -> they are both not accessible in the help browser. That's mostly fine for heroes.
> 
>   -> the main unit's help page is still accessible on the map, the variation has still the same problem as before
> 
> 
> 
> Using hide_help=no for both:
> 
>   -> the variation's help entry can now be accessed over the map, but also over the help and
> 
>   -> due to the "variations:" line in the main units help entry this would spoil the player
> 
> 
> 
> For completeness sake (or rather, to show another issue)
> 
> Using hide_help=yes only for the main unit:
> 
>   -> Isn't useful here
> 
>   -> the variation has usually a link to the base unit. That link is now displayed in red and broken. It should just not be displayed
> 
> 
> 
> 
> 
> TL;DR (again)
> 
> I think things would be better
> 
>   -> if variations with hide_help=yes would have an help entry which is only accessible over the map (like for unit_types)
> 
>   -> if the link to the main unit is not shown when the main unit uses hide_help=yes

**Comment by wesnoth-bugs** [comment]
**Modified on 2017-03-24**

**wedge009 changed title:** help: -> help & [variation]s: effect of the hide_help key

**Comment by Wedge009** [comment]
@sevu There's a lot in this old report so I'm not sure I'm understanding it properly. Is it related at all to #4142? I think I've seen other Help issues relating to variations but they've since been fixed.

**Comment by jostephd** [comment]
@Wedge009 #4142 is unrelated; it's just about the `Profile` button in the `Attack enemy` dialog opening the wrong help page. The point of this report is that after Konrad gets the sceptre, if you select him on the map and press `d` the help shows the sprite without the sceptre, not the same sprite as in the map.

I'm not sure I agree that this is a problem, though. It's described in implementation terms, but players don't know whether "Konrad has a new ranged attack and a different sprite" is implemented as a variation or as a pickup-able item, like the storm trident, which _isn't_ shown in the help. In fact, the case could be made that showing the ranged attack in the help would be confusing, since it would mean the help for, say, the `Commander` unit type will list different attacks depending on what point in the campaign it is opened from.

As an alternative, I think a version of the unit type help pages that show the sprites, portraits, description and statistics of a particular unit, as modified by whatever [object]s that unit may have, would solve this feature request too. It would mean the help page for `Commander` would stay as it is but there would be a way to show the statistics of Konrad, complete with whatever objects he may have (for example, Sceptre of Fire and Void Armor, if he has both of them). This was discussed before here: https://github.com/wesnoth/wesnoth/issues/1091#issuecomment-299754450 @sevu, would this work for you?

**Comment by Pentarctagon** [comment]
Is this something that should be kept open, or closed at this point?

**Comment by Wedge009** [comment]
I think @sevu was the original reporter.

**Comment by Wedge009** [comment]
I don't think I really understood the original issue and @sevu has yet to respond - perhaps busy.

**Comment by Wedge009** [comment]
Closing due to age and lack of response.

**Comment by CelticMinstrel** [comment]
I suspect that this is technically real but not quite the same as the original report – I _think_ at some point it became impossible to view _any_ hidden topics even when explicitly loaded from the map, so in Sevu's example of both having hide_help=yes, there would be no way to view the unit's help page.

Probably best to check and open a new issue if that's true though.
